src/pages/iou/IOUCurrencySelection.js

```diff
@@ -56,125 +56,98 @@ const defaultProps = {
     },
 };
 
-class IOUCurrencySelection extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            searchValue: '',
-            currencyData: this.getCurrencyOptions(this.props.currencyList),
-        };
-        this.getCurrencyOptions = this.getCurrencyOptions.bind(this);
-        this.getSections = this.getSections.bind(this);
-        this.confirmCurrencySelection = this.confirmCurrencySelection.bind(this);
-        this.changeSearchValue = this.changeSearchValue.bind(this);
-    }
-
-    /**
-     * Returns the sections needed for the OptionsSelector
-     *
-     * @returns {Array}
-     */
-    getSections() {
-        if (this.state.searchValue.trim() && !this.state.currencyData.length) {
+function IOUCurrencySelection(props) {
+    const selectedCurrencyCode = lodashGet(props.route, 'params.currency', props.iou.selectedCurrencyCode, CONST.CURRENCY.USD);
+    const currencyOptions = useMemo(
+        () =>
+            _.map(props.currencyList, (currencyInfo, currencyCode) => {
+                const isSelectedCurrency = currencyCode === selectedCurrencyCode;
+                return {
+                    text: `${currencyCode} - ${CurrencyUtils.getLocalizedCurrencySymbol(currencyCode)}`,
+                    currencyCode,
+                    keyForList: currencyCode,
+                    customIcon: isSelectedCurrency ? greenCheckmark : undefined,
+                    boldStyle: isSelectedCurrency,
+                };
+            }),
+        [selectedCurrencyCode, props.currencyList],
+    );
+
+    const confirmCurrencySelection = useCallback(
+        (option) => {
+            const backTo = lodashGet(props.route, 'params.backTo', '');
+            // When we refresh the web, the money request route gets cleared from the navigation stack.
+            // Navigating to "backTo" will result in forward navigation instead, causing disruption to the currency selection.
+            // To prevent any negative experience, we have made the decision to simply close the currency selection page.
+            if (_.isEmpty(backTo) || props.navigation.getState().routes.length === 1) {
+                Navigation.goBack();
+            } else {
+                Navigation.navigate(`${props.route.params.backTo}?currency=${option.currencyCode}`);
+            }
+        },
+        [props.route, props.navigation],
+    );
+
+    const [searchValue, setCurrentSearchValue] = useState('');
```

Comment:
let's clean up a little bit - useState's logic should be define on the top of the function